pkg/addonmanager/controllers/agentdeploy/controller.go

```diff
@@ -219,8 +219,14 @@ func (c *addonDeployController) sync(ctx context.Context, syncCtx factory.SyncCo
 	}
 
 	if err = c.updateAddon(ctx, addon, oldAddon); err != nil {
+		if errors.IsConflict(err) {
+			syncCtx.Queue().AddRateLimited(key)
+			return nil
```

Comment:
if return nil,  https://github.com/open-cluster-management-io/addon-framework/blob/main/pkg/basecontroller/factory/base_controller.go#L167  this place will forget the key.
I think it may be better to check if conflict  below https://github.com/open-cluster-management-io/addon-framework/blob/549a39952992436903c51bf3849d8530271a1654/pkg/basecontroller/factory/base_controller.go#L158

Reply:
OK, if we handle conflict error in `base_controller` then every client controller will get the behavior as well. I haven't looked at all the other controllers using `base_controller` but if you think there won't be any issues then I'll `base_controller`.

Reply:
yes I think we could log error when log level is 4 and it is conflict error